### ISSUE TITLE
EF remove entity when use automapper to update entity properties

### ISSUE THREAD

**Issue by singlewind** [dotnet/efcore#9156]
Describe what is not working as expected.

I use following code to update an entity in the database

```C#
var currentMed = await context.Media.FindAsync(srcMed.Id);
mapper.Map(srcMed, currentMed);
await context.SaveChangesAsync();
```

This approach works in one context but not another. I think it was the AutoMapper mapping cause this. What condition make EF to delete an entity except call delete method.

Funny thing is when I get 
```C#
var entry = await context.Entry(currentMed);
```
the state of entry is actually Modified not Deleted.

### Further technical details
EF Core version: 1.1.2
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017

**Comment by ajcvickers** [comment]
**EF Team Triage:** This issue is lacking enough information for us to be able to effectively investigate it. In particular, it is missing the following information requested in the new issue template. Can you please provide this information?

> ### Steps to reproduce
**Ideally include a complete code listing that we can run to reproduce the issue.
Alternatively, you can provide a project/solution that we can run.**

Note that this does not have to be your actual production code--in fact the more it can be simplified the better--but it must demonstrate the issue you are seeing.

---

_BTW we're not just doing this to be mean :smile:... we get a lot traffic on this project and it takes time to attempt to reproduce an issue based on fragments of information. In addition, our attempt is often unsuccessful as the exact conditions required to hit the issue are often not explicitly included in the code provided. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we ask that folks give us a self-contained way to reproduce an issue._

_For a guide on submitting good bug reports, read [Painless Bug Tracking](http://www.joelonsoftware.com/articles/fog0000000029.html)._

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by singlewind** [comment]
Hi @ajcvickers, thank you for your reply. I think I figured out what is the issue. It was my fault. 

I have a foreign key [ForeignId] and also [ForeignObject] in this class. When I map the object. I only map the [ForeignId] and leave the [ForeignObject] null which cause the object to be removed from database.

**Comment by sergei66666** [comment]
I faced the same problem.
@singlewind say "When I map the object. I only map the [ForeignId] and leave the [ForeignObject] null which cause the object to be removed from database."

I want add some info. Behavior depends on [ForeignObject] value before map.
1) If it is null BEFORE map all works fine. We get update;
2) If not null and become null after map we get delete.

In any case we belive that [ForeignId] has value.

In other words. If [ForeignObject] changed from some value to null - entity will be deleted. Despite the fact that [ForeignId] remains the same.

Here similar questions
[https://stackoverflow.com/questions/47665106/entity-framework-deleting-object-upon-update](url)
[https://stackoverflow.com/questions/47334441/entity-framework-core-2-0-deleting-record-instead-of-updating-it](url)
